### PR TITLE
Adding auth trigger to pubsub

### LINKED ISSUE #1273: Provide trigger authentication support for GCP scalers
GCP should support TriggerAuthentication like other scalers

### Use-Case
This is more secure

Would be happy to take this issue up with little guidance. 
I see that gcp scaler is reading auth creds from config.TriggerMetadata . So if I change it to read from config.Auth , will it just work or are there any other changes that I would need to do?

### PR DESCRIPTION
**Pull request by harshthakur9030**
Signed-off-by: Harsh Thakur <harshthakur9030@gmail.com>

_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Tests have been added
- [x] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs . https://github.com/kedacore/keda-docs/pull/293
- [x] Changelog has been updated

Relates to #1273

### CODE REVIEW

**Review by ahmelsayed — APPROVED** [review]

**Comment by ahmelsayed** [comment]
LGTM. Thanks @harshthakur9030!

**Comment by tomkerkhove** [comment]
Would you mind updating our docs on https://github.com/kedacore/keda-docs/blob/master/content/docs/2.0/scalers/gcp-pub-sub.md as well please?

**Review by tomkerkhove — CHANGES_REQUESTED** [review]
Can you update the changelog please?
https://github.com/kedacore/keda/blob/v2/CHANGELOG.md

File: pkg/scalers/gcp_pubsub_scaler_test.go [hunk 1]
```diff
@@ -19,17 +20,21 @@ type gcpPubSubMetricIdentifier struct {
...
-	// missing credentials
-	{map[string]string{"subscriptionName": "mysubscription", "subscriptionSize": "7", "credentialsFromEnv": ""}, true},
+	{nil, map[string]string{"subscriptionName": "mysubscription", "subscriptionSize": "7", "credentialsFromEnv": "SAMPLE_CREDS"}, false},
+	// // missing subscriptionName
+	{nil, map[string]string{"subscriptionName": "", "subscriptionSize": "7", "credentialsFromEnv": "SAMPLE_CREDS"}, true},
+	// // missing credentials
+	{nil, map[string]string{"subscriptionName": "mysubscription", "subscriptionSize": "7", "credentialsFromEnv": ""}, true},
 	// incorrect credentials
-	{map[string]string{"subscriptionName": "mysubscription", "subscriptionSize": "7", "credentialsFromEnv": "WRONG_CREDS"}, true},
+	{nil, map[string]string{"subscriptionName": "mysubscription", "subscriptionSize": "7", "credentialsFromEnv": "WRONG_CREDS"}, false},
```
Comment (zroubalik) [inline pkg/scalers/gcp_pubsub_scaler_test.go:31]: As you mentioned in the slack, this should be revised, we don't have to keep this test case, as it is the same as the one on line 25

**Review by zroubalik — APPROVED** [review]
LGTM, just a minor nit in tests and as Tom mentioned, please add that to docs and add this to Changelog (Improvement section)

**Comment by harshthakur9030** [comment]
PTAL @zroubalik  @tomkerkhove ! Did the changes asked. Here's the doc PR: https://github.com/kedacore/keda-docs/pull/293

**Review by tomkerkhove — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

**Pull request merged**